Add handlers for location as NavSatFixes to/from ROS

LoyVanBeek opened the pull request:
Send and receive locations in Telegram as NavSatFix in ROS. 
NavSatFix seemed the most relevant and best message type for this, because it is after all a NavSatFix if you share your location through Telegram base don your phone's GPS. 

Another benefit of NavSatFix is that this does not add more dependencies

Commit:
Add handlers for location as NavSatFixes to/from ROS

Timple commented:
Haha, nice!

MatthijsBurgh commented on scripts/telegram_ros_bridge:
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
+        :param msg: NavSatFix that the robot wants to share
+        """
+        if not self._telegram_chat_id:
+            rospy.logerr("ROS Bridge not initialized, dropping string message")
```
String message, copy paste error

LoyVanBeek committed:
String-->location in copy-pasted handler

reinzor commented on scripts/telegram_ros_bridge:
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
+    def _telegram_location_callback(self, _, update):
+        """
+        Called when a new telegram Location is received. The method will verify whether the incoming Location is
+        from the bridges telegram conversation by comparing the chat_id.
```
typo, bridges --> bridged

reinzor commented on scripts/telegram_ros_bridge:
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
+        :param update: Received update that holds the chat_id and message data
+        """
+        location = update.message.location
+        if isinstance(location, Location):
```
Why this check, log if we receive something else?

LoyVanBeek replied:
I removed the check, it's useless now

reinzor commented on scripts/telegram_ros_bridge:
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
+        if isinstance(location, Location):
+            rospy.loginfo("Incoming location from telegram: {}".format(location))
+            if self._telegram_chat_id is None:
+                rospy.logwarn("Discarding message. No active chat_id.")
```
Duplicated code .. Maybe we should refactor this in another PR

LoyVanBeek replied:
https://github.com/tue-robotics/telegram_ros/issues/2

reinzor commented on scripts/telegram_ros_bridge:
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
+                    "ROS Bridge initialized to another chat_id. Type /start to connect to this chat_id")
+            else:
+                navsatfix = NavSatFix()
+                navsatfix.header.stamp = update.message.date # TODO convert tot ROS Time
```
Can be removed I guess

LoyVanBeek replied:
Hey, I thought I fixed this. I do want to include the timestamp the user sent the location

LoyVanBeek replied:
Ah, I see now, done correctly in the lines below. 

I made all this on the train with laptop battery dying so i had to be quick :-)

reinzor commented on scripts/telegram_ros_bridge:
```diff
@@ -141,6 +149,46 @@ class TelegramROSBridge(object):
+                navsatfix.header.stamp = update.message.date # TODO convert tot ROS Time
+                sec_since_epoch = (update.message.date - datetime.datetime(1970,1,1)).total_seconds()
+                navsatfix.header.stamp = rospy.Time.from_sec(sec_since_epoch)
+                navsatfix.header.frame_id = "telegram"
```
"frame_id is the frame of reference reported by the satellite, usually the location of the antenna.", this does not make sense. Better leave it empty if unknown.

reinzor commented:
Cool PR :)

LoyVanBeek committed:
Do not set frame_id and don't check for Location

We don't know the frame_id really and the check is unneeded if the update Filter is working as it should.

Commit:
Remove obsolete TODO for header stamp setting

reinzor committed:
Satisfy pep8 linter

reinzor reviewed (approved):
PR seems fine to me. Did not test this so I assume the functionality is working.

LoyVanBeek commented:
I did of course test this, works quite nicely. We're going to use this to share the location of a little autonomous boat :-)
